### PR TITLE
6.patch/api/reviews/review

### PR DESCRIPTION
**Pull request by mini-wizbit**

### CODE REVIEW

**Comment by mini-wizbit** [comment]
Did merge but only this branch to fix errors I have done the first set of feedback changing code but had to close this pull request first then resend... I will get the hang of this!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -129,6 +129,152 @@ describe("Using app.js to run the database of NC-games", () => {
...
+          expect(body.review).toHaveProperty("review_id", 7);
+          expect(body.review).toHaveProperty(
+            "title",
+            "Mollit elit qui incididunt veniam occaecat cupidatat"
+          );
+          expect(body.review).toHaveProperty("category", "social deduction");
+          expect(body.review).toHaveProperty("designer", "Avery Wunzboogerz");
+          expect(body.review).toHaveProperty("owner", "mallionaire");
+        });
+    });
```
Comment (emilyb93) [inline __tests__/app.test.js]: This is the same test as the one above, we can remove one of these

File: __tests__/app.test.js [hunk 2]
```diff
@@ -129,6 +129,152 @@ describe("Using app.js to run the database of NC-games", () => {
...
+          expect(body.review).toHaveProperty(
+            "title",
+            "Occaecat consequat officia in quis commodo."
+          );
+          expect(body.review).toHaveProperty("category", "social deduction");
+          expect(body.review).toHaveProperty("designer", "Ollie Tabooger");
+          expect(body.review).toHaveProperty("owner", "mallionaire");
+        });
+    });
+    test("ERROR handling... what if the review_id is 9999", () => {
```
Comment (emilyb93) [inline __tests__/app.test.js:199]: Love the suspense... here

File: Model/reviews.patch.model.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      }
+      if (validReviewKeys.includes(toPatchThisKey[0])) {
+        queryStr += ` SET ${toPatchThisKey[0]} = $1`;
+        queryStr += ` WHERE review_id = $2 RETURNING *;`;
+      }
+      createdQueryObj.string = queryStr;
+      createdQueryObj.value = newValue;
+      createdQueryObj.id = id;
+      return createdQueryObj;
+    })
```
Comment (emilyb93) [inline Model/reviews.patch.model.js]: We dont really need to do a check for the article first as we can simply do an UPDATE operation with a RETURNING and then check how many rows are returned to us. If we get rows.length === 0, then we know the article doesnt exist. This saves us doing 2 queries for this. Also passing the wrong data type into and SQL string will make SQL throw an error based on that so we dont really need to do a lot of this checking in JS

File: Controllers/reviews.post.controller.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  const review_id = req.params;
+  const patchInfo = req.body;
+  selectReviewPost(review_id, patchInfo)
+    .then((updatedData) => {
+      res.status(201).send({ updatedData });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
```
Comment (emilyb93) [inline Controllers/reviews.post.controller.js:13]: Why are we talking about POST requests for a patch request? These functions are never used, remove them

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Why are we talking about POST requests for a patch request? These functions are never used, remove them

File: __tests__/app.test.js (hunk 1 above)
Comment (mini-wizbit) [reply __tests__/app.test.js]: Okay, I did the second test to be above the simpler test, however if you believe they are the same then all is good.

File: __tests__/app.test.js (hunk 2 above)
Comment (mini-wizbit) [reply __tests__/app.test.js:199]: I KNOW RIGHT!!!

File: Controllers/reviews.post.controller.js (hunk 4 above)
Comment (mini-wizbit) [reply Controllers/reviews.post.controller.js:13]: there should be patch folders with the correction as Poonam did call me out for that last time

File: Model/reviews.patch.model.js (hunk 3 above)
Comment (mini-wizbit) [reply Model/reviews.patch.model.js]: ToDo less bulk and look for a sql code of 'abc' then let the custom error handling the rest...

**Review by emilyb93 — APPROVED** [review]
Looks good!

**Pull request merged**